config/routes.rb

```diff
@@ -0,0 +1,22 @@
+Rails.application.routes.draw do
+  # resources :tasks, only: [:index, :show]
+  # get "/tasks", to: "tasks#index"
+  get "/tasks", to: "tasks#index", as: "tasks"
+
+  get "tasks/new", to: "tasks#new", as: "new_task"
+  post "/tasks", to: "tasks#create"
+
+  get "/tasks/:id", to: "tasks#show", as: "task"
+
+  get "/tasks/:id/edit", to: "tasks#edit", as: "edit_task"
+  patch "/tasks/:id", to: "tasks#update"
+
+  patch "/tasks/:id/mark_complete", to: "tasks#mark_complete", as: "mark_complete"
```

Comment:
Just a note, that you don't need 2 routes to toggle the task between complete and incomplete, you could do it with one.